File: src/ConsistentStrings.java
import java.util.HashSet;
import java.util.Set;

public class ConsistentStrings {
    public static int countConsistentStrings(String allowed, String[] words) {
        Set<Character> allowedChars = new HashSet<>();
        for (char c : allowed.toCharArray()) {
            allowedChars.add(c);
        }

        int count = 0;
        for (String word : words) {
            boolean isConsistent = true;
            for (char c : word.toCharArray()) {
                if (!allowedChars.contains(c)) {
                    isConsistent = false;
                    break;
                }
            }
            if (isConsistent) {
                count++;
            }
        }
        return count;
    }
    
    public static void main(String[] args) {
        // Ejemplo 1
        String allowed1 = "ab";
        String[] words1 = {"ad", "bd", "aaab", "baa", "badab"};
        System.out.println(countConsistentStrings(allowed1, words1)); // Output: 2

        // Ejemplo 2
        String allowed2 = "abc";
        String[] words2 = {"a", "b", "c", "ab", "ac", "bc", "abc"};
        System.out.println(countConsistentStrings(allowed2, words2)); // Output: 7

        // Ejemplo 3
        String allowed3 = "cad";
        String[] words3 = {"cc", "acd", "b", "ba", "bac", "bad", "ac", "d"};
        System.out.println(countConsistentStrings(allowed3, words3)); // Output: 4
    }
}

File: easy/Python/ConsistentStrings.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: list[str]) -> int:
        allowed_set: set[str] = set(allowed)
        consistent_count: int = 0

        for word in words:
            if all(char in allowed_set for char in word):
                consistent_count += 1

        return consistent_count

solution: Solution = Solution()
allowed: str = "ab"
words: list[str] = ["ad", "bd", "aaab", "baa", "badab"]
result: int = solution.countConsistentStrings(allowed, words)
print(result) # 2
